### PR TITLE
Destination-bigquery: to handle new datatypes

### LINKED ISSUE #17647: Update destination-bigquery to handle the new data types protocol
See https://github.com/airbytehq/airbyte/issues/17637. These destinations rely on the same JsonToAvroSchemaConverter as destination-s3, so we shouldn't tackle this issue until after finishing https://github.com/airbytehq/airbyte/issues/17645.

They do have some extra logic on top of it though, so they won't be completely free.

TODO (@edgao ): add info on how destination connector will have to declare supported protocol version

### PR DESCRIPTION
**Pull request by etsybaev**
This PR is based on the protocol v1 branch https://github.com/airbytehq/airbyte/pull/20036

## What
A new datatype protocol is going to be introduced. So need to make sure that the connector will handle it properly.

## How
Most of fixes were done in the Normalization branch. Basically, the bigquery connector handles new protocol out of the box. Minor changes were done in recently added integrationTests to make all tests pass. I had to temporarily comment some test records in PR to make it pass. Previously we didn't have it and not all of them may be processed by normalization properly. Those will be fixed (implemeted) as part of other tasks. 

Currently, integration tests fail because of secrets mismatch. Will be fixed once this PR https://github.com/airbytehq/airbyte/pull/20851 is merged 

## Recommended reading order
This PR is based on unmerged yet Normalization branch. So check the only files in commit related to bigquery PR https://github.com/airbytehq/airbyte/pull/20898/commits/dc410921c885d39167d35903b70b321fb00780c8

## 🚨 User Impact 🚨
Breaking changes. May only be merged once other stuff related to new datatype protocol is done

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by grishick** [comment]
@etsybaev I merged https://github.com/airbytehq/airbyte/pull/20851. Please rebase and re-test this PR.

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_basic_test_messages.txt
Comment (edgao) [inline airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_basic_test_messages.txt:17]: what's the purpose of changing these emitted_at timestamps?

**Comment by edgao** [comment]
fyi I changed the base branch of this PR to  `etsybaev/17642-updated-normalization-to-new-datatypes` for ease of review

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_array_test_catalog.json
Comment (edgao) [inline airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_array_test_catalog.json:9]: json schema so confusing :(

**Review by edgao — APPROVED** [review]
just to confirm: the diff is correct, and I didn't screw up anything with git? i.e. it's the data type test catalog+message changes, plus destination-bigquery declaring protocol v1 in the DAT?

one small question, but otherwise lgtm assuming the branch is still in good state!

File: airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_basic_test_messages.txt
Comment (etsybaev) [reply airbyte-integrations/bases/standard-destination-test/src/main/resources/v1/data_type_basic_test_messages.txt:17]: In some of DBs we relay on emittedAt inserting order.  So if EmittedAt does not correspond to the order in test file - then the test will compare the wrong values.

**Comment by sherifnada** [comment]
@edgao can you clarify why extensibility review is needed here? I don't think we changed any code owned by the team in this PR

**Comment by edgao** [comment]
I think I pushed some commits to the wrong branch :( doing some messy git rebases right now, sorry for the mistag.

**Comment by edgao** [comment]
ah. this is what happens when you force-push a base branch >.> 

neat

**Comment by edgao** [comment]
this PR is now based on the superbranch (https://github.com/airbytehq/airbyte/pull/20036). Leaving this as draft for now, because once that branch is merged to master, this PR will have a super weird-looking diff. I _think_ merging master back into this branch will fix, at which point we can re-mark as ready for review.

**Comment by edgao** [comment]
this needs to get updated from master - 99% of the diff is because we just merged the (previous) base branch into master.

**Comment by evantahler** [comment]
I guess we can close this now?

**Comment by edgao** [comment]
yep, closing.

**Pull request closed without merging**
